src/lib.rs

```diff
@@ -307,19 +307,43 @@ fn get_num_cpus() -> usize {
 fn get_num_cpus() -> usize {
     use std::ptr;
 
+    let mut cpus: libc::c_uint = 0;
+    let mut cpus_size = std::mem::size_of_val(&cpus);
+    let mut mib = [libc::CTL_HW, libc::HW_NCPUONLINE, 0, 0];
+    let rc: libc::c_int;
+
+    unsafe {
+        rc = libc::sysctl(mib.as_mut_ptr(),
+                          2,
+                          &mut cpus as *mut _ as *mut _,
+                          &mut cpus_size as *mut _ as *mut _,
+                          ptr::null_mut(),
+                          0);
+    }
+    if rc < 0 {
+        cpus = 1;
+    }
+    cpus as usize
+}
+
+#[cfg(target_os = "openbsd")]
+fn get_num_physical_cpus() -> usize {
+    use std::ptr;
+
     let mut cpus: libc::c_uint = 0;
     let mut cpus_size = std::mem::size_of_val(&cpus);
     let mut mib = [libc::CTL_HW, libc::HW_NCPU, 0, 0];
+    let rc: libc::c_int;
 
     unsafe {
-        libc::sysctl(mib.as_mut_ptr(),
-                     2,
-                     &mut cpus as *mut _ as *mut _,
-                     &mut cpus_size as *mut _ as *mut _,
-                     ptr::null_mut(),
-                     0);
+        rc = libc::sysctl(mib.as_mut_ptr(),
+                          2,
+                          &mut cpus as *mut _ as *mut _,
+                          &mut cpus_size as *mut _ as *mut _,
+                          ptr::null_mut(),
+                          0);
     }
-    if cpus < 1 {
+    if rc < 0 {
```

Comment:
Is there never a case where `sysctl` could succeed, but leave the number at 0?

Reply:
Not according to the [manpage](https://man.openbsd.org/sysctl.2#RETURN_VALUES) which explicitly states:

> If the call to sysctl() is unsuccessful, -1 is returned and errno is set appropriately.